projects/storefrontlib/src/cms-components/myaccount/order/replenishment-order-history/replenishment-order-history.component.html

```diff
@@ -0,0 +1,222 @@
+                  {{ 'orderHistory.replenishmentOrderId' | cxTranslate }}
+                </div>
+                <a
+                  [routerLink]="
+                    {
+                      cxRoute: 'replenishmentDetails',
+                      params: order
+                    } | cxUrl
+                  "
+                  class="cx-replenishment-order-history-value"
+                >
+                  {{ order?.replenishmentOrderCode }}</a
+                >
+              </td>
+              <td
+                class="cx-replenishment-order-history-placed"
+                (click)="goToOrderDetail(order)"
+              >
+                <div class="d-md-none cx-replenishment-order-history-label">
+                  {{ 'orderHistory.purchaseOrderNumber' | cxTranslate }}
+                </div>
+                <a
+                  [routerLink]="
+                    {
+                      cxRoute: 'replenishmentDetails',
+                      params: order
+                    } | cxUrl
+                  "
+                  class="cx-replenishment-order-history-value"
+                >
+                  {{
+                    order?.purchaseOrderNumber?.length > 0
+                      ? order?.purchaseOrderNumber
+                      : ('orderHistory.emptyPurchaseOrderId' | cxTranslate)
+                  }}
+                </a>
+              </td>
+              <td
+                class="cx-replenishment-order-history-status"
+                (click)="goToOrderDetail(order)"
+              >
+                <div class="d-md-none cx-replenishment-order-history-label">
+                  {{ 'orderHistory.startOn' | cxTranslate }}
+                </div>
+                <a
+                  [routerLink]="
+                    {
+                      cxRoute: 'replenishmentDetails',
+                      params: order
+                    } | cxUrl
+                  "
+                  class="cx-replenishment-order-history-value"
+                >
+                  {{ order?.firstDate | cxDate }}</a
+                >
+              </td>
+              <td
+                class="cx-replenishment-order-history-total"
+                (click)="goToOrderDetail(order)"
+              >
+                <div class="d-md-none cx-replenishment-order-history-label">
+                  {{ 'orderHistory.frequency' | cxTranslate }}
+                </div>
+                <a
+                  [routerLink]="
+                    {
+                      cxRoute: 'replenishmentDetails',
+                      params: order
+                    } | cxUrl
+                  "
+                  class="cx-replenishment-order-history-value"
+                >
+                  {{ order?.trigger.displayTimeTable }}
+                </a>
+              </td>
+              <td
+                class="cx-replenishment-order-history-total"
+                (click)="goToOrderDetail(order)"
```

Comment:
I see that you made it this way to handle the problem where when you cancel, it would change page due to the event propagation. 

I'll keep this in check as maybe we can do something better.

Reply:
I did some digging and found that adding click event listeners to every `td` is expensive. 
Imagine order history has x amount of columns (not realistic) but adding x amount of click event listeners is not good. You can also imagine if one day the history is display in a set of rows instead of columns, x amount of click event won't help the application.

---
Remember the call we had when we talked about this. I believe you literally had the answer as you brought out event propagation. What you could've done is keep the click event in the tr and when you click the cancel button, you call the event.stopPropagation to stop it from propagating. 

so please fix

Reply:
-----
another thing is is a reminder since it was said long time so I did not expect you to remember. But when building features from old accelerator, we want to most of the time match the UX. 

when you cancel in the history page, it no longer shows the cancel button and either the first date or next date says cancelled (will get back to this second point once I get access again as I'm re-initializing at the moment of writing this)

Reply:
changed made to add a click on the `tr` for details view and click on the cancel button for the modal

Reply:
ux suggested with having `td` for the cancle button to go order details page

Reply:
yup that's what I said too earlier as I don't believe having the `td` open the cancel modal was a correct choice.

anyway, no need to put in the `td` as the whole tr will be set to it